geoips/create_plugin_registries.py

```diff
@@ -344,10 +370,22 @@ def add_yaml_plugin(filepath, relpath, package, plugins):
                 for subplg_name in subplg_names:
                     if str(subplg_name[0]) not in list(plugins[interface_name].keys()):
                         plugins[interface_name][str(subplg_name[0])] = {}
+                    pd = None
+                    if "product_defaults" in list(yaml_subplg.keys()):
+                        pd = yaml_subplg["product_defaults"]
+                    family = None
+                    if "family" in list(yaml_subplg.keys()):
+                        family = yaml_subplg["family"]
                     plugins[interface_name][str(subplg_name[0])][
                         str(subplg_name[1])
                     ] = {
+                        "docstring": yaml_subplg["docstring"],
+                        "family": family,
+                        "interface": interface_module.name,
                         "package": plugin["package"],
+                        "plugin_type": "yaml_based",
+                        "product_defaults": pd,
+                        "source_names": yaml_subplg["source_names"],
                         "relpath": plugin["relpath"],
```

Comment:
@evrose54  Does this resolve the docstring propagation between product_defualts and products ?

Reply:
This portion grabs the docstring from the sub-product plugins, or the top level docstring of any other type of yaml plugin. That's why we got an error in data fusion, as some sub-product plugins in 'layered.yaml' did not have docstrings. In general, I think it is appropriate to just use the docstrings of all products, rather than their product defaults, as they are also included in the registries.

Reply:
So will this use "products" docstring if it exists, and "product_defaults" docstring if the docstring doesn't exist in the product itself?  In which case we would not have to make any changes to the data_fusion package to get it to work?

Reply:
No, it will just use the docstring from a plugin. I meant that we could find the corresponding docstring in the registry manually, however we can't guarantee that it will be included by the time we process the product plugin. That's why it's important that every product has a docstring, otherwise a KeyError will occur and those products won't be included in the registry.

Reply:
I think there needs to be logic to look in product_defaults for any missing fields in a given product when creating the registry.  So the default docstring would be included in the registry itself when it's created.

Reply:
I modified this to do exactly that in the most recent update to this branch.

Reply:
Indeed you did: https://github.com/NRLMMD-GEOIPS/geoips/pull/392/commits/b30fb53f1dd0985e557fcfb229cec8836ab367be